src/context.ml

```diff
@@ -190,10 +197,30 @@ let create ~(kind : Kind.t) ~path ~env ~name ~merlin ~targets () =
     in
 
     let build_dir = Path.of_string (sprintf "_build/%s" name) in
+    let ocamlpath =
+      match
+        let var = "OCAMLPATH" in
+        match kind, findlib_toolchain with
+        | Default, None -> Env.get env var
+        | _ ->
+          (* If we are not in the default context, we can only use the
+             OCAMLPATH variable if it is specific to this build
+             context *)
+          (* CR-someday diml: maybe we should actually clear OCAMLPATH
+             in other build contexts *)
+          match Env.get env var, Env.get (Env.initial ()) var with
+          | None  , None   -> None
+          | Some s, None   -> Some s
+          | None  , Some _ -> None
+          | Some x, Some y -> Option.some_if (x <> y) x
```

Comment:
Why do we discard the OCAMLPATH if it's set in the context? Is it because we're afraid it's simply set by the parent process? I think in this case, opam should simply unset when giving us the environment variables.

Reply:
Yes. I guess opam should do it indeed, but currently it doesn't. Additionally, for cross-compilation we don't rely on opam.